### PR TITLE
Export command should raise exception on failure

### LINKED ISSUE #2487: Export command should give non-zero exit code on failure
**Problem:**

When no environment id is given, the export command fails with the following log lines:

```
stdout: inmanta.export           WARNING Empty deployment model.
inmanta.export           ERROR   The environment for this model should be set!
inmanta.export           INFO    Committed resources with version 0
```

However, the exit code is zero, which indicates success.

**Tasks:**

- [ ] Investigate the impact of returning a non-zero exit code in case of failure (server-side compile, lsm, etc.)
- [ ] Ensure that a non-zero exit code is returned when the environment id is not set.
- [ ] Document exit codes 
- [ ] Ensure lsm still works

### PR DESCRIPTION
**Pull request by andraskvr**
# Description

Other failures (than the missing environment) were already not silent

closes #2487 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] ~End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: src/inmanta/export.py [hunk 1]
```diff
@@ -441,7 +441,7 @@ def commit_resources(self, version: int, resources: List[Dict[str, str]], metada
         tid = cfg_env.get()
         if tid is None:
             LOGGER.error("The environment for this model should be set!")
-            return
+            raise Exception("The environment for this model should be set!")
```
Comment (wouterdb) [inline src/inmanta/export.py:444]: This is not exactly the most specific exception type, but it is in line with the rest of the method

File: tests/test_app_cli.py [hunk 2]
```diff
@@ -336,3 +357,67 @@ def other_exporter(exporter: Exporter) -> None:
...
+    process = await subprocess.create_subprocess_exec(*args, stdout=sys.stdout, stderr=sys.stderr)
+    try:
+        await asyncio.wait_for(process.communicate(), timeout=30)
+    except asyncio.TimeoutError as e:
+        process.kill()
+        await process.communicate()
+        raise e
+
+    # Make sure exitcode is one
+    assert process.returncode == 1, f"Process ended with bad return code, got {process.returncode} (expected 1)"
```
Comment (wouterdb) [inline tests/test_app_cli.py:417]: I would like to see this exact number fixed in the testcase. (the return code number is part of the API)

**Review by wouterdb — CHANGES_REQUESTED** [review]

File: tests/test_app_cli.py (hunk 2 above)
Comment (andraskvr) [reply tests/test_app_cli.py:417]: @wouterdb I'm not sure I understand, do you mean a specific number that corresponds to the missing environment error?

**Review by sanderr — APPROVED** [review]

File: tests/test_app_cli.py (hunk 2 above)
Comment (wouterdb) [reply tests/test_app_cli.py:417]: ah no, I read the code wrong, this is fine.

**Review by wouterdb — APPROVED** [review]

**Pull request merged**